src/net.cpp

```diff
@@ -1819,13 +1823,19 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
                 continue;
 
             // only connect to full nodes
-            if ((addr.nServices & REQUIRED_SERVICES) != REQUIRED_SERVICES)
+            if ( ((addr.nServices & NODE_NETWORK) != NODE_NETWORK)
+                 && ((addr.nServices & NODE_NETWORK_LIMITED) != NODE_NETWORK_LIMITED)) {
                 continue;
+            }
 
             // only consider very recently tried nodes after 30 failed attempts
             if (nANow - addr.nLastTry < 600 && nTries < 30)
                 continue;
 
+            // only consider nodes matching alternative and missing relevant services after 10 failed attempts and only if less than a fourth of the outbound are up.
+            if ( ((addr.nServices & nAlternativeServices) == nAlternativeServices) && (addr.nServices & nRelevantServices) != nRelevantServices && (nTries < 20 || nOutbound >= (nMaxOutbound >> 2)))
```

Comment:
Seems that we would never connect to limited nodes because of the code on lines 1840-1847. `nRequiredServices` is set to either `nRelevantServices` or `REQUIRED_SERVICES`, never `nAlternativeServices`.